MNE/disp3D/adapters/ecdview.cpp

```diff
@@ -0,0 +1,152 @@
+
+#include <QGridLayout>
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace DISP3DLIB;
+using namespace INVERSELIB;
+using namespace MNELIB;
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// DEFINE MEMBER METHODS
+//=============================================================================================================
+
+ECDView::ECDView(const DipoleFitSettings& dipFitSettings, const ECDSet& ecdSet, QWidget* parent)
+: QWidget(parent)
+, m_p3DView(View3D::SPtr(new View3D()))
+, m_pData3DModel(Data3DTreeModel::SPtr(new Data3DTreeModel()))
+{
+    //Init 3D View
+    m_p3DView->setModel(m_pData3DModel);
+
+    QStringList slControlFlags;
+    slControlFlags << "Data" << "View" << "Light";
+    m_pControl3DView = Control3DWidget::SPtr(new Control3DWidget(this, slControlFlags));
+
+    m_pControl3DView->init(m_pData3DModel, m_p3DView);
+
+    //Read mri transform
+    QFile file(dipFitSettings.mriname);
+    ECDSet ecdSetTrans = ecdSet;
+
+    if(file.exists()) {
+        FiffCoordTrans coordTrans(file);
+
+        std::cout << std::endl << "coordTrans" << coordTrans.trans;
+        std::cout << std::endl << "coordTransInv" << coordTrans.invtrans;
+
+        for(int i = 0; i < ecdSet.size() ; ++i) {
+            MatrixX3f dipoles(1, 3);
+            //transform location
+            dipoles(0,0) = ecdSet[i].rd(0);
+            dipoles(0,1) = ecdSet[i].rd(1);
+            dipoles(0,2) = ecdSet[i].rd(2);
+
+            dipoles = coordTrans.apply_trans(dipoles);
+
+            ecdSetTrans[i].rd(0) = dipoles(0,0);
+            ecdSetTrans[i].rd(1) = dipoles(0,1);
+            ecdSetTrans[i].rd(2) = dipoles(0,2);
+
+            //transform orientation
+            dipoles(0,0) = ecdSet[i].Q(0);
+            dipoles(0,1) = ecdSet[i].Q(1);
+            dipoles(0,2) = ecdSet[i].Q(2);
+
+            dipoles = coordTrans.apply_trans(dipoles);
+
+            ecdSetTrans[i].Q(0) = dipoles(0,0);
+            ecdSetTrans[i].Q(1) = dipoles(0,1);
+            ecdSetTrans[i].Q(2) = dipoles(0,2);
+        }
+    }
+
+    //Add ECD data
+    ECDSet::SPtr pEcdSetTrans = ECDSet::SPtr(new ECDSet(ecdSetTrans));
+    m_pData3DModel->addDipoleFitData("sample", QString("Set %1").arg(dipFitSettings.setno), pEcdSetTrans);
+
+    //Read and show BEM
+    QFile t_fileBem(dipFitSettings.bemname);
+
+    if(t_fileBem.exists()) {
+        MNEBem t_Bem(t_fileBem);
+        m_pData3DModel->addBemData("sample", "BEM", t_Bem);
+    }
```

Comment:
when file not exists throw a warning that no bem can be displayed due to missing  bem model